CMakeLists.txt

```diff
@@ -93,6 +93,14 @@ if( USE_CUDA )
     list( APPEND HIP_INCLUDE_DIRS "${HIP_ROOT_DIR}/include" )
 endif( )
 
+
+option(BUILD_CODE_COVERAGE "Build with code coverage enabled" OFF)
+if(BUILD_CODE_COVERAGE)
+  add_compile_options(-fprofile-arcs -ftest-coverage)
```

Comment:
In rocBLAS it looks like we're defaulting to use debug mode if this option is passed in, probably want the same here.